### ISSUE TITLE
in functions param SetLength calculate not correct len

### ISSUE THREAD

**Issue by xscrable** [wmnsk/go-m3ua#17]
example:
```
func (a *AspUpAck) SetLength() {
	if param := a.AspIdentifier; param != nil {
		param.SetLength()
	}
	if param := a.InfoString; param != nil {
		param.SetLength()
	}

	a.Header.SetLength() //8 byte header
	a.Header.Length += uint32(a.MarshalLen()) // MarshalLen return len payload+len header
}

**Comment by wmnsk** [comment]
Thank you.
If I understand what you pointed out correctly, `SetLength` sets longer length in `Length` field when a message has any payload(=when `SetLength` is called manually), correct?
If so, I think PR #18 fixes your concern.

**Comment by wmnsk** [comment]
RFC4666 says Message Length field should include the length of Common Header. So, adding `-8` seems invalid.

https://tools.ietf.org/html/rfc4666#section-3.1.4

FYI, here's the packets that the examples in this repo actually generates. At least Wireshark does not say it's wrong.

[go-m3ua_example.pcapng.gz](https://github.com/wmnsk/go-m3ua/files/4492016/go-m3ua_example.pcapng.gz)

**Comment by xscrable** [comment]
current code
```
func (a *AspUpAck) SetLength() {
	if param := a.AspIdentifier; param != nil {
		param.SetLength()
	}
	if param := a.InfoString; param != nil {
		param.SetLength()
	}

	a.Header.SetLength()
	a.Header.Length += uint32(a.MarshalLen())
}

// MarshalLen returns the serial length of AspUpAck.
func (a *AspUpAck) MarshalLen() int {
	l := 8
	if param := a.AspIdentifier; param != nil {
		l += param.MarshalLen()
	}
	if param := a.InfoString; param != nil {
		l += param.MarshalLen()
	}
	return l
}

MarshalLen return minimum len 8 bytes (len header), a.Header.SetLength() return 8bytes -> we have len 16 bytes! that is, the header length is counted twice

**Comment by wmnsk** [comment]
Ah OK, thanks. I got what's wrong now. Let me fix it in PR#18 later.

**Comment by wmnsk** [comment]
Fixed in cf34e37. I removed redundant `a.Header.SetLength()` instead of adding `-8`.
Here's the pcap generated by the example with that version.

[go-m3ua_example_pr18.pcapng.gz](https://github.com/wmnsk/go-m3ua/files/4492331/go-m3ua_example_pr18.pcapng.gz)
